mh/session.py

```diff
@@ -116,37 +177,73 @@ def delete(self):
          - We should probably create a SessionManager thread per server.
         """
         if not self.request_reconnection:
-            DB.delete_session(self)
+            STATE.delete_session(self)
 
     def is_jap(self):
         """TODO: Heuristic using the connection data to detect region."""
         pass
 
     def new_pat_ticket(self):
-        DB.new_pat_ticket(self)
+        STATE.new_pat_ticket(self)
         return to_bytearray(self.pat_ticket)
 
+    def get_fmp_version(self):
+        return STATE.get_servers_version()
+
     def get_users(self, first_index, count):
-        return DB.get_users(self, first_index, count)
+        return STATE.get_users(self, first_index, count)
 
     def use_user(self, index, name):
-        DB.use_user(self, index, name)
+        STATE.use_user(self, index, name)
+
+    def server_index_exists(self, index):
+        try:
+            if index in self.loaded_server_ids:
+                self.recall_server(index)
+                return True
+            else:
+                return False
+        except KeyError as _:
```

Comment:
The `index` is guaranteed to exist in the `loaded_server_ids`, but the corresponding `server_id` is not guaranteed to exist in the `Cache`'s `self.servers` list. Actually, I use an `assert` in the function that's used here to check for that (`other/cache.py` line 407), so I should be excepting an `AssertionError` rather than a `KeyError`.

This has been fixed in accordance with that.